app/resources/api/v2/labware_resource.rb

```diff
@@ -4,31 +4,23 @@ module Api
   module V2
     # LabwareResource
     class LabwareResource < BaseResource
-      attributes :uuid
-      attribute :created_at
+      # We import most labware shared behaviour, this includes associations,
+      # attributes and filters. By adding behaviour here we ensure that it
+      # is automatically available on plate and tube.
+      include Api::V2::SharedBehaviour::Labware
 
       default_includes :uuid_object, :barcodes
 
-      has_one :purpose, readonly: true, foreign_key: :plate_purpose_id
-      has_one :custom_metadatum_collection
-      has_many :comments, readonly: true
-      has_many :direct_submissions, readonly: true
-      has_many :state_changes, readonly: true
-
-      has_many :ancestors, readonly: true, polymorphic: true
-      has_many :descendants, readonly: true, polymorphic: true
-      has_many :parents, readonly: true, polymorphic: true
-      has_many :children, readonly: true, polymorphic: true
-
-      filter :purpose_name,
-             apply:
-               (lambda { |records, value, _options| records.joins(:purpose).where(plate_purposes: { name: value }) })
-      filter :purpose_id, apply: ->(records, value, _options) { records.where(plate_purpose_id: value) }
-      filter :barcode, apply: ->(records, value, _options) { records.with_barcode(value) }
-      filter :uuid, apply: ->(records, value, _options) { records.with_uuid(value) }
-      filter :without_children, apply: ->(records, _value, _options) { records.without_children }
-      filter :created_at_gt,
-             apply: (lambda { |records, value, _options| records.where('labware.created_at > ?', value[0].to_date) })
+      # Custom methods
+      # These shouldn't be used for business logic, and a more about
+      # I/O and isolating implementation details.
+      def labware_barcode
+        {
+          'ean13_barcode' => _model.try(:ean13_barcode),
+          'machine_barcode' => _model.try(:machine_barcode),
+          'human_barcode' => _model.try(:human_barcode)
+        }
+      end
```

Comment:
Initially I tried to actually add these methods to labware, by including barcodable, but the change was not pleasant and it was clearly impacting behaviour throughout SS.